code/data_structures/src/list/singly_linked_list/operations/insertion/insertion_at_nth_position.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+  C++ program to insert element at the nth position of a Singly Linked List
+  Part of Cosmos by OpenGenus Foundation
+ */
+#include<iostream>
+#include<cstdlib>
+
+using namespace std;
+
+
+struct node{
+             int data;
+             node *next;
+
+            };
+
+void printList(node *head)
+{
+    cout<<"This is your current List\n";
+    while(head!=NULL)
+    {
+        cout<<head->data<<" => ";
+        head=head->next;
+    }
+    cout<<"NULL\n";
+}
+
+void insertAtTail(node **head,int data)
+{
+    node *newptr = new node;
+    newptr->data=data;
+    newptr->next=NULL;
+
+    node *dummy = *head;
+    while(dummy->next!=NULL) dummy=dummy->next;
+    dummy->next = newptr;
+    return;
+}
+
+//function to insert node at nth position of the Linked List
+void insertAtPosition(node **head,int position,int data)
+{
+    node *newptr=new node;
+    newptr->data=data;
+
+    node *dummy = *head;                //dummy variable is used to traverse to the required position
+    node *prev = dummy;                 //prev node to keep track of the previous node
+
+    while(--position)                   //loop to traverse to the required position
```

Comment:
You must do a check to make sure `position` is between 0 and length of the list that is a valid value.